e2e/nomostest/config_sync.go

```diff
@@ -892,34 +875,30 @@ func setupCentralizedControl(nt *NT, opts *ntopts.New) {
 		ns := nn.Namespace
 		rsCount++
 		rsNamespaces[ns] = struct{}{}
+
+		// Add a namespace for the RepoSync.
 		nt.RootRepos[configsync.RootSyncName].Add(StructuredNSPath(ns, ns), fake.NamespaceObject(ns))
-		rb := RepoSyncRoleBinding(nn)
-		dependencies := []client.Object{cr, rb}
-		nt.RootRepos[configsync.RootSyncName].Add(StructuredNSPath(ns, fmt.Sprintf("rb-%s", nn.Name)), rb)
-		if isPSPCluster() {
-			// Add a ClusterRoleBinding so that the pods can be created
-			// when the cluster has PodSecurityPolicy enabled.
-			// Background: If a RoleBinding (not a ClusterRoleBinding) is used,
-			// it will only grant usage for pods being run in the same namespace as the binding.
-			// TODO: Remove the psp related change when Kubernetes 1.25 is
-			// available on GKE.
-			crb := repoSyncClusterRoleBinding(nn)
-			dependencies = append(dependencies, crb)
-			nt.RootRepos[configsync.RootSyncName].Add(fmt.Sprintf("acme/cluster/crb-%s-%s.yaml", ns, nn.Name), crb)
-		}
 
+		// Add a ClusterRoleBinding to grant the RepoSync reconciler permission
+		// to manage resources in the RepoSync namespace.
```

Comment:
I can see how this makes the dependencies/cleanup easier to manage, however I don't see this as being realistic. Switching to a ClusterRoleBinding means that the RepoSync permissions are being granted for the whole cluster rather than being restricted to the RepoSync's namespace.

It doesn't seem to me like this is something we would recommend to users, since RepoSyncs are intended to be limited in scope to a single namespace. I feel that we should come up with a solution to make this work with RoleBindings - WDYT?

Reply:
Hmm. For some reason I thought ClusterRoleBinding could apply to a namespace, but you're right, they apply to all namespaces... That means our existing PSP workaround is a big security risk, and so is this PR...

Reply:
The only way I can think of to avoid deadlock when deleting a namespace is to make a controller that injects finalizers on objects with an incoming dependency to block their deletion until the objects that depend on them are deleted. This would enforce deletion ordering on all clients.

Sounds like a fair amount of work tho...

Reply:
bug filed with k8s (tho we'll want a faster fix locally): https://github.com/kubernetes/kubernetes/issues/115070